### PR TITLE
Disallow changes to distance setting after creating a class

### PR DESCRIPTION
**Pull request by nilskulawiak**
### What's being changed:
Disallows changing the distance parameter of a class after it has been created. This should not be possible, because the HNSW graph depends on it. Closes https://github.com/weaviate/weaviate/issues/3177.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation: Not necessary
- [ ] Chaos pipeline run or not necessary. Link to pipeline: Not necessary
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary. Not necessary

### CODE REVIEW

File: adapters/repos/db/vector/hnsw/config_update.go [hunk 1]
```diff
@@ -58,18 +62,18 @@ func ValidateUserConfigUpdate(initial, updated schema.VectorIndexConfig) error {
 	return nil
 }
 
-type immutableInt struct {
-	accessor func(c ent.UserConfig) int
+type immutableParameter struct {
```
Comment (antas-marcin) [inline adapters/repos/db/vector/hnsw/config_update.go:65]: why this struct was changed from `immutableInt` to `immutableParameter`? is it used somewhere else?

Comment (nilskulawiak) [reply adapters/repos/db/vector/hnsw/config_update.go:65]: `distance` is of type string, which is why the accessor now returns an `interface{}` instead on an int. `immutableParameter` is therefore a more fitting name. And no this is not used anywhere else to my knowledge.

File: adapters/repos/db/vector/hnsw/config_update.go [hunk 2]
```diff
@@ -58,18 +62,18 @@ func ValidateUserConfigUpdate(initial, updated schema.VectorIndexConfig) error {
...
+	accessor func(c ent.UserConfig) interface{}
 	name     string
 }
 
-func validateImmutableIntField(u immutableInt,
+func validateImmutableIntField(u immutableParameter,
```
Comment (aliszka) [inline adapters/repos/db/vector/hnsw/config_update.go:70]: function have `Int` in its name, which does not match argument type anymore

**Review by aliszka — APPROVED** [review]

**Review by antas-marcin — APPROVED** [review]

**Comment by antas-marcin** [comment]
@nilskulawiak in order for us to be able to merge your PR, you need to agree to our CLA. Simply adding a comment `I agree to CLA` will enable us to merge your PR

**Pull request merged**